### ISSUE TITLE
Custom notification sound does not play if phone was sleeping for a few minutes

### ISSUE THREAD

**Issue by tanuva** [signalapp/Signal-iOS#4910]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
My custom notification sound is not played if a message arrives after my phone was sleeping (locked, screen off) for a few minutes.

### Steps to reproduce
- Configure a custom notification sound in Signal's notification settings (chosen from files, a plain wav file, about 1.5 seconds long, 230 KB)
- Receive a message with the phone locked and sleeping for a few minutes

**Actual result:** No notification sound is played, yet the screen lights up and shows a notification from Signal.

If I wait for the screen to go dark again and send another message, it arrives silently again.

If I unlock my phone (via Face ID, without doing anything else) and receive another message then, the notification sound is played as expected. If I send another message immediately after the screen goes dark from an unlocked state, the sound is still played. Only after a few minutes it reverts to the failing behaviour again.

**Expected result:** The notification sound is played together with a notification appearing on screen.

### Device info

**Device**: iPhone 12 mini

**iOS version**: 14.4

**Signal version:** 5.6.0.14

### Link to debug log

https://debuglogs.org/1565a548a3690e3969b7f9a087fc66738fbe454640fa38497b2ae09752a47ad4.zip

Steps taken:
1. Enabled debug log
2. Immediately locked the phone
3. Waited 2-3 minutes
4. Received a silent message
5. Unlocked the phone and submitted the log

**Comment by alopix** [comment]
I’m experiencing the exact same issue. Custom sound works perfectly fine when phone is in use but not being played once sleeping.

**Comment by tanuva** [comment]
This is still reproducible with iOS 15.0.2.

**Comment by tanuva** [comment]
This is still reproducible with iOS 15.2.1, still with the same instructions as originally given.

> Is there anything you can do to help move it forward?
Well, it would be nice if *you* could at least confirm if you can reproduce the problem, too. :)

**Comment by NiklasBr** [comment]
Why does Signal even use Stale Bot if no-one is paying attention?

**Comment by alopix** [comment]
This is still the case in iOS 15.4

**Comment by tanuva** [comment]
*Still* relevant on iOS 15.4.1. I'm slowly getting frustrated here. The lack of communication isn't helping.

**Comment by NiklasBr** [comment]
What use is Stale Bot if no-one in the org is paying attention anyway?

**Comment by tanuva** [comment]
It tries to get rid of us but it wasn‘t very successful so far, except at building up resentment.

**Comment by NiklasBr** [comment]
What use is Stale Bot if no-one in the org is paying attention anyway?

**Comment by tanuva** [comment]
Oi, this hasn’t moved forward since two years because no developer was interested in it, not because we didn’t provide information. If I can’t reopen I‘ll have to file a new issue with the same request. Wtf, stale-bot?!
